### PR TITLE
Upgrade `emojibase` and `twemoji`

### PR DESCRIPTION
**Pull request by SimonBrandner**
Type: task

<hr>

Fixes emoji for `:D` and allows emoji for capital `:P`

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://61e5413fc5b56620e97d4035--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Review by t3chguy — APPROVED** [review]
So I don't think we can merge this because Twemoji hasn't been updated to Emoji 14 yet https://github.com/twitter/twemoji/releases. But LGTM otherwise.

File: src/components/views/emojipicker/EmojiPicker.tsx [hunk 1]
```diff
@@ -195,8 +195,11 @@ class EmojiPicker extends React.Component<IProps, IState> {
     };
 
     private emojiMatchesFilter = (emoji: IEmoji, filter: string): boolean => {
-        return emoji.annotation.toLowerCase().includes(filter) ||
-            emoji.emoticon?.toLowerCase().includes(filter) ||
+        return emoji.label.toLowerCase().includes(filter) ||
+            (Array.isArray(emoji.emoticon)
+                ? emoji.emoticon.some((x) => x.includes(filter))
+                : emoji.emoticon.includes(filter)
```
Comment (tulir) [inline src/components/views/emojipicker/EmojiPicker.tsx:201]: image

```suggestion
                : emoji.emoticon?.includes(filter)
```

Comment (SimonBrandner) [reply src/components/views/emojipicker/EmojiPicker.tsx]: Thanks for the catch!

**Comment by tulir** [comment]
This seems to also break autocompleting emojis with non-prefix strings, e.g. :monocle_face: (`:face_with_monocle:`) by typing `:monocle`. I have no idea why/how though, maybe something to do with the emojibase-regex update

**Comment by tulir** [comment]
I figured out the problem, the fields used by emoji autocompletion aren't type-checked:

```diff
diff --git a/src/autocomplete/EmojiProvider.tsx b/src/autocomplete/EmojiProvider.tsx
index 960b0f9475..55d58a8953 100644
--- a/src/autocomplete/EmojiProvider.tsx
+++ b/src/autocomplete/EmojiProvider.tsx
@@ -75,7 +75,7 @@ export default class EmojiProvider extends AutocompleteProvider {
             shouldMatchWordsOnly: false,
         });
         this.nameMatcher = new QueryMatcher(SORTED_EMOJI, {
-            keys: ['emoji.annotation'],
+            keys: ['emoji.label'],
             // For removing punctuation
             shouldMatchWordsOnly: true,
         });
```

**Review by t3chguy — COMMENTED** [review]
Wonder if we can prove the typing here using something like https://stackoverflow.com/questions/45415436/nested-keyof-object-paths-using-dot-notation

**Comment by SimonBrandner** [comment]
Twemoji has now been upgraded, so after a review this should be ready to go

**Review by t3chguy — APPROVED** [review]
Looks good to me, just pending an oob manual test of the new Twemoji fonts

**Comment by SimonBrandner** [comment]
After going through the emoji, it seems nothing is broken

**Pull request merged**